core/src/multi_proof.rs

```diff
@@ -0,0 +1,548 @@
+//! Generate a multiproof from a vector of path proofs.
+//! The multiproof will contain the minimum information needed to verify
+//! the inclusion of all provided path proofs.
+
+use crate::{
+    proof::{PathProof, PathProofTerminal},
+    trie::Node,
+};
+use bitvec::{order::Msb0, view::BitView};
+
+// Each terminal node may have a set of siblings that are uniquely
+// used during the verification with its terminal node.
+// This struct contains the terminal node itself,
+// the depth at which the siblings start to be uniquely associated to that terminal,
+// and the siblings themselves.
+#[derive(Debug, Clone)]
+struct SubPathProof {
+    terminal: PathProofTerminal,
+    depth: u8,
```

Comment:
does this suffer from the same problem as trie_pos, where we might want depth = 256 and it could be out of range for u8?

maybe better to use a usize

Reply:
yes, you're totally right. There could be a case where depth is 256, for example if two keys with only the lsb as different bit then there would be two `TraverseInfo` with `path_bit_index` equal to 256 but it would not be used as index for the key_path because by construction those `TraverseInfo` would represent a single `path_proof` and not a range.  

I have added also another test to cover this scenario `test_multiproof_creation_two_path_proofs_256_depth`